rotkehlchen/tests/api/test_historical_balances.py

```diff
@@ -0,0 +1,208 @@
+from http import HTTPStatus
+from typing import TYPE_CHECKING
+
+import pytest
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.constants import DAY_IN_SECONDS
+from rotkehlchen.constants.assets import A_BTC, A_ETH
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.base import HistoryEvent
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.tests.utils.api import (
+    api_url_for,
+    assert_error_response,
+    assert_proper_sync_response_with_result,
+)
+from rotkehlchen.types import Location, Timestamp
+from rotkehlchen.utils.misc import ts_sec_to_ms
+
+if TYPE_CHECKING:
+    from rotkehlchen.api.server import APIServer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+START_TS = Timestamp(1672531200)
+
+
+def create_test_events(db: 'DBHandler') -> None:
+    events = [
+        HistoryEvent(  # Day 1: Initial receive
+            event_identifier='btc_receive_1',
+            sequence_index=0,
+            timestamp=ts_sec_to_ms(START_TS),
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            location=Location.BLOCKCHAIN,
+            asset=A_BTC,
+            balance=Balance(amount=FVal('2')),
+            notes='Receive BTC',
+        ),
+        HistoryEvent(  # another receive
+            event_identifier='eth_receive_1',
+            sequence_index=1,
+            timestamp=ts_sec_to_ms(Timestamp(START_TS + 3000)),
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            location=Location.BLOCKCHAIN,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('10')),
+            notes='Receive ETH',
+        ),
+        HistoryEvent(  # Day 3: Partial spend
+            event_identifier='btc_spend_1',
+            sequence_index=2,
+            timestamp=ts_sec_to_ms(Timestamp(START_TS + DAY_IN_SECONDS * 2)),
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.NONE,
+            location=Location.BLOCKCHAIN,
+            asset=A_BTC,
+            balance=Balance(amount=FVal('0.5')),
+            notes='BTC partial spend',
+        ),
+    ]
+
+    with db.user_write() as write_cursor:
+        DBHistoryEvents(database=db).add_history_events(
+            write_cursor=write_cursor,
+            history=events,
+        )
+
+
+@pytest.fixture(name='setup_historical_data')
+def fixture_setup_historical_data(rotkehlchen_api_server: 'APIServer') -> None:
+    """Fixture to setup historical test data"""
+    db = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
+    create_test_events(db=db)
```

Comment:
Ah wait if this is only used here, just move the function contents here in the fixture